packages/scripts/src/admin-proposals/utils.js

```diff
@@ -247,6 +253,84 @@ const verifyGovernanceRootTunnelMessage = async (targetAddress, message, governo
+    const votingToken = new web3.eth.Contract(ExpandedERC20.abi, await _getContractAddressByName("VotingToken", 1));
+
+    // Caller must have 5000 voting token in wallet and have approved proposer contract to pull them.
+    const votingTokenBalance = toBN(await votingToken.methods.balanceOf(caller).call());
+    const requiredVotingTokenBalance = toBN(await proposer.methods.bond().call());
+    if (votingTokenBalance.lt(requiredVotingTokenBalance)) {
+      throw new Error(
+        `Caller only has ${votingTokenBalance.toString()} UMA, but submitting a proposal requires ${requiredVotingTokenBalance.toString()} UMA.`
+      );
+    }
+    const votingTokenAllowance = toBN(await votingToken.methods.allowance(caller, proposer.options.address).call());
+    if (votingTokenAllowance.lt(requiredVotingTokenBalance)) {
+      console.log(`- Approving proposer @ ${proposer.options.address} to spend unlimited UMA from caller...`);
+      const approvalTxn = await votingToken.methods
+        .approve(proposer.options.address, MAX_UINT_VAL)
+        .send({ from: caller, ...gasPriceObj });
+      console.log(`- Transaction: ${approvalTxn?.transactionHash}`);
+    }
+
+    const txn = await proposer.methods.propose(adminProposalTransactions).send({ from: caller, ...gasPriceObj });
+    console.log("- Transaction: ", txn?.transactionHash);
+
+    // Print out details about new Admin proposal
+    const priceRequests = await oracle.getPastEvents("PriceRequestAdded");
+    const newAdminRequest = priceRequests[priceRequests.length - 1];
+    console.log(
+      `- New admin request {identifier: ${
+        newAdminRequest.returnValues.identifier
+      }, timestamp: ${newAdminRequest.returnValues.time.toString()}}`
+    );
+  } else {
+    console.log("- 0 Transactions in Admin proposal. Nothing to do");
+  }
+  console.groupEnd();
+};
+
+const resolveProposals = async (web3, caller, proposalsToLookback = 10) => {
+  const proposer = new web3.eth.Contract(Proposer.abi, await _getContractAddressByName("Proposer", 1));
+  const governor = new web3.eth.Contract(Governor.abi, await _getContractAddressByName("Governor", 1));
+
+  console.group(`\nResolving the latest ${proposalsToLookback} proposals`);
+  const totalProposals = Number(await governor.methods.numProposals().call());
+  const bondedProposals = await Promise.all(
+    [...Array(totalProposals).keys()].map((i) => proposer.methods.bondedProposals(i).call())
+  );
+  for (let i = bondedProposals.length - 1; i >= bondedProposals.length - 1 - proposalsToLookback; i--) {
+    const bondedProposal = bondedProposals[i];
+    if (toBN(bondedProposal.lockedBond.toString()).gt(toBN(0))) {
+      console.log(
+        `- Proposal #${i} has a locked bond of ${bondedProposal.lockedBond.toString()} UMA, proposer was @ ${
+          bondedProposal.sender
+        }`
+      );
+      try {
+        const txn = await proposer.methods.resolveProposal(i).send({ from: caller });
```

Comment:
Could be faster and more gas efficient to use multicall here (we could simulate the txn before adding it to the group of txns to send with multicall). Not a must-have though, so we could add this later.